app/renderer/js/utils/domain-util.ts

```diff
@@ -1,17 +1,14 @@
 import {JsonDB} from 'node-json-db';
 
-import escape from 'escape-html';
-import request from 'request';
 import fs from 'fs';
 import path from 'path';
 import Logger from './logger-util';
-import {remote} from 'electron';
+import {remote, ClientRequestConstructorOptions} from 'electron';
 
-import * as RequestUtil from './request-util';
 import * as EnterpriseUtil from './enterprise-util';
 import * as Messages from '../../../resources/messages';
 
-const {app, dialog} = remote;
+const {app, dialog, net, session} = remote;
```

Comment:
Unfortunately I don’t think `electron.net` can be used safely via `remote`; see electron/electron#12545. I think we’ll need to build our own IPC interface and use it in the main process.

Reply:
So it would be better to move all `electron.net` requests to the main process via a IPCRenderer/Main interface call?

Following this up, we could have a separate module in the main process that handles the replacement of remote calls in the renderer process? Solving #915 .

Reply:
Okay, I was facing the `end` problem too. Seems plausible. Thanks @andersk .
Also, 
> IPC primitives (`ipcRenderer` and `ipcMain`) will be unaffected. This only affects the `remote` module, which is a convenience layer wrapped on top of those IPC primitives.
> 
> The recommended migration path for apps is to replace usage of the remote module with explicit use of the IPC primitives, for example [`ipcRenderer.invoke`](https://electronjs.org/docs/api/ipc-renderer#ipcrendererinvokechannel-args) / [`ipcMain.handle`](https://electronjs.org/docs/api/ipc-main#ipcmainhandlechannel-listener).
>

Is it safe to say that the the `primitives` are safer and better, performance-wise, to use and hence to migrate to?
Are there any cases where they lag behind `remote`?